src/ProtoSchema.ts

```diff
@@ -19,7 +19,9 @@ export default class ProtoSchema implements Schema {
     if (!parent) throw new ConfluentSchemaRegistryArgumentError('no nested fields')
     const keys = Object.keys(parent)
     const reflection = parent[keys[0]]
-    if (reflection instanceof Namespace && reflection.nested)
+
+    // Traverse down the nested Namespaces until we find a message Type instance (which extends Namespace)
+    if (reflection instanceof Namespace && !(reflection instanceof Type) && reflection.nested)
```

Comment:
Open Question: Is it enough to just check that it's not a Type and has a nested object?
```javascript
if(!(reflection instanceof Type) && reflection.nested)
```

Reply:
I tried the patch as per @MikeDalby and it helped me for a nested ProtoBuf. Went from 
```
  decodedValue: MealItems { itemName: 'pizza' }
```
to a much better
```
  decodedValue: Meal {
    item: [],
    drink: [ [DrinkItems], [DrinkItems] ],
    name: 'pizza'
  }
```
Producer was this Python snippet:
```
mybeer = meal_pb2.Meal.DrinkItems(drink_name="beer")
mywine = meal_pb2.Meal.DrinkItems(drink_name="wine")

meal = meal_pb2.Meal(name='pizza', drink=[mybeer,mywine])
# Less meal (for testing kafkajs which seems to miss the drinks
#meal = meal_pb2.Meal(name='pizza', drink=[])

producer.produce(topic=topic, key=str(uuid4()), value=meal)
```
While I'm missing the "beer" and "wine", at least I got drinks now.